lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -233,13 +256,45 @@ def _check_size():
             return self.size == keysize
 
         self.fingerprint = fingerprint
-        self.public_key = pubkey
 
         if not perms_required:
             return _check_state() and _check_type() and _check_size()
 
         return _check_state() and _check_perms(module) and _check_type() and _check_size()
 
+    def isPublicKeyValid(self, module):
+
+        def _get_pubkey_content():
+            if os.path.exists(self.path + ".pub"):
+                with open(self.path + ".pub", "r") as pubkey_f:
+                    present_pubkey = pubkey_f.read().strip('\n')
+                return present_pubkey
+            else:
+                return False
+
+        def _pubkey_valid(pubkey):
+            if _get_pubkey_content():
+                present_pubkey = _get_pubkey_content()
```

Comment:
I don't think it is a good idea to call `_get_pubkey_content()` multiple times. Better call it once and store the result, so it can be used both in `_pubkey_valid` and `_comment_valid`.

Reply:
Done.